File: backend/src/main/java/com/gfi/parkplatzapp/backend/persistence/entities/Mitarbeiter.java
package com.gfi.parkplatzapp.backend.persistence.entities;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.ArrayList;
import java.util.List;

@Entity
@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
public class Mitarbeiter {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long mitarbeiterID;
    private String uid;

    private String vorname;
    private String nachname;
    private String mail;
    @OneToMany
    private List<Verstoss> verstossList;
    @OneToMany
    private List<Kennzeichen> kennzeichenList;

}

File: frontend/src/app/facade/Mitarbeiter.ts
import { Kennzeichen } from "./Kennzeichen"
import { Verstoss } from "./Verstoss";

export interface Mitarbeiter {
    
    mitarbeiterID: number;
    vorname: string;
    nachname: string;
    mail: string;
    kennzeichenList: Kennzeichen[];
    verstossList: Verstoss[];
};

